.github/workflows/python-docs-theme.yml

```
@@ -0,0 +1,118 @@
+name: Update and pull python-docs-theme project
+
+on:
+  push:
+    branches:
+      - 'main'
+    paths-ignore:
+      - 'docs/**'
+      - '.readthedocs.yaml'
+      - 'sample-workflows/**'
+  pull_request:
+    branches:
+      - 'main'
+    paths-ignore:
+      - 'docs/**'
+      - '.readthedocs.yaml'
+      - 'sample-workflows/**'
+  workflow_dispatch:
+    inputs:
+      push:
+        description: "Push source strings and translations to Transifex"
+        type: boolean
+      pull:
+        description: "Pull translations from Transifex"
+        type: boolean
+  schedule:
+    - cron: '0 0 * * 1'
+
+env:
+  TX_CLI_VERSION: '1.6.16'
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+
+    steps:
+      - name: Check out ${{ github.repository }}
+        uses: actions/checkout@v4.2.2
+
+      - name: Check out python-docs-theme
+        uses: actions/checkout@v4.2.2
+        with:
+          repository: python/python-docs-theme
+          persist-credentials: false
+          path: python-docs-theme
+
+      - name: Set up Python 3
+        uses: actions/setup-python@v5.6.0
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+          cache-dependency-path: |
+            requirements.txt
+            python-docs-theme/requirements.txt
+
+      - name: Install Transifex CLI
+        working-directory: /usr/local/bin
+        run: |
+          curl -o- https://raw.githubusercontent.com/transifex/cli/master/install.sh | bash -s -- v${{ env.TX_CLI_VERSION }}
+
+      - name: Install dependencies
+        run: |
+          sudo apt update -y && sudo apt install gettext -y
+          pip3 install --upgrade pip
+          pip3 install -r requirements.txt -r python-docs-theme/requirements.txt
+
+        # required for generating .tx/config
+      - name: Generate POT files
+        working-directory: python-docs-theme
+        run: |
+          python babel_runner.py extract
+
```

Let's store POT as well? Maybe this could be added to the above step.

```suggestion

      - name: Store the POT locally
         run: |
           cp ../python-docs-theme/python_docs_theme/locale/python-docs-theme.pot .

```